File: frontend/src/model/Ticket.ts
import { Formal } from "./Formal";
import { TicketRequest } from "./TicketRequest";

// TODO: fix this
export interface FormalTicket {
    formal: Formal;
    ticket: Ticket;
    guestTickets: Ticket[];
}

export interface Ticket {
    id: string;
    isGuest: boolean;
    isQueue: boolean;
    option: string;
    formalId: string;
    userId: string;
}

export type AdminTicket = Ticket & {
    userName: string;
    userEmail: string;
}

File: backend/model/ticket.go
package model

import "github.com/google/uuid"

type Ticket struct {
	Model
	IsGuest    bool      `json:"isGuest"`
	IsQueue    bool      `json:"isQueue"`
	MealOption string    `json:"option"`
	FormalID   uuid.UUID `json:"formalId"`
	Formal     *Formal   `json:"-" gorm:"constraint:OnDelete:CASCADE;"`
	UserID     uuid.UUID `json:"userId"`
	User       *User     `json:"-" gorm:"constraint:OnDelete:CASCADE;"`
}
